terraso_backend/audit_logs/models.py

```diff
@@ -0,0 +1,49 @@
+from django.db import models
+
+
+# Create your models here.
+class Log(models.Model):
+    """
+    Log model for audits logs
+    """
+    timestamp = models.DateTimeField(auto_now_add=True)
+    keyValueCache = models.JSONField()
+    client_timestamp = models.DateTimeField()
+
+    def __str__(self):
+        return str(self.client_timestamp) + " - " + self.keyValueCache
+
+    def get_string(self, template: str = None) -> str:
+        if template is None:
+            return str(self)
+        return template.format(**self.keyValueCache)
+
+
+# TODO - Discuss: if we should just store string for the the log resource or
+# foreing keys if we store foreing key they will have to be generic keys
+class KeyValue(models.Model):
```

Comment:
The advantage is essential that the definition may change, and the change itself. It was a small inspiration in open telemetry; it could also fulfill a metadata role, we can discuss how much we want to use it or not, and we could also have fk to a helpful resource and other to a generic resource.  That will also be a valid alternative.

Reply:
i see. IMO, for our use case, having to change the schema is a good safety check so that we don't accidentally corrupt the audit log table.